### PR TITLE
Initial setup for kaiju-plugin-utils

### PR DESCRIPTION
**Pull request by mjhenkes**
This package offers common kaiju plugin utils for using webpack to generate previews. Basic config and a virtual filesystem that will fall back to the real file system if a file isn't found.

The readme needs to be updated before this pull request is approved.

### CODE REVIEW

**Review by StephenEsser — APPROVED** [review]

File: tests/jest/data/package.json [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/mh015940/source/kaiju-plugin-utils/package.json
```
Comment (StephenEsser) [inline tests/jest/data/package.json:1]: image

Comment (mjhenkes) [reply tests/jest/data/package.json:1]: That does look shady, this is supposed to be a symlink. I'm not sure how well files like this translate into git, but the test using the file still passes. Which might be an indicator of the quality of test. 

https://github.com/cerner/kaiju-plugin-utils/pull/1/files#diff-81255a28f27e8cd84fb6ab3fc0eceffaR13

File: README.md [hunk 2]
```diff
@@ -1 +1,94 @@
...
+The path the the root of the kaiju node server. Uses the app-root-path library.
+
+**Return values:** A string representing the root path.
+
+## Fallback File System
+The fallback file system is a very specific proxy setup on top of a memory file system. This filesystem falls back to the supplied file system when a file isn't found in the memory file system. This was created to allow webpack to use the kaiju servers node modules instead of requiring an npm install on the in memory file system to generate the preview. The fs methods that will 'fallback' are readFileSync, statSync, readlinkSync, existsSync and readdirSync. These were chosen based on webpacks usage.
+
+### Usage
+```js
+const {, fallbackFs} = require('kaiju-plugin-utils');
```
Comment (StephenEsser) [inline README.md]: Extra comma

File: README.md [hunk 3]
```diff
@@ -1 +1,94 @@
...
+
+### runCompiler
+Runs the webpack compiler on the files included in the memory file system with the config supplied.
+
+**Arguments:**
+Memory File System - A memory file system containing the files to be webpacked.
+
+Config - The webpack config to be used by the compiler.
+
+**Return values:** Compiler file system - A promise for a memory file system contained the output from the webpack compiler.
```
Comment (StephenEsser) [inline README.md]: The description might be need to be updated. Might read cleaner swapping `contained`  to `containing` in `memory file system contained`

File: src/FallbackFileSystem.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        };
+      }
+      return origMethod;
+    },
+  };
+
+  return new Proxy(new MemoryFS(data), handler);
+}
+
+// export default fallbackFS;
```
Comment (StephenEsser) [inline src/FallbackFileSystem.js]: Do you want to keep the ES6 syntax comment?

File: README.md (hunk 2 above)
Comment (mjhenkes) [reply README.md]: The extra comma is intended, we're using destructuring to include just fallbackFs.

File: src/FallbackFileSystem.js (hunk 4 above)
Comment (mjhenkes) [reply src/FallbackFileSystem.js]: It should be deleted.

File: README.md (hunk 2 above)
Comment (mjhenkes) [reply README.md]: I was wrong (for the better), will be fixing.

**Review by StephenEsser — APPROVED** [review]

**Pull request merged**